toothpick-runtime/src/main/java/toothpick/ScopeImpl.java

```diff
@@ -78,6 +80,7 @@ public ScopeImpl(Object name) {
 
   @Override
   public <T> Lazy<T> getLazy(Class<T> clazz, String name) {
+    crashIfInvalid();
```

Comment:
as you want. Though having in the scope implementation makes more sense conceptually.